src/common/strtol.cc

```diff
@@ -16,18 +16,30 @@
 
 #include <climits>
 #include <limits>
+#include <cmath>
 #include <sstream>
+#include <string_view>
 
 using std::ostringstream;
 
-long long strict_strtoll(const char *str, int base, std::string *err)
+long long strict_strtoll(const std::string_view str, int base, std::string *err)
 {
-  char *endptr;
   std::string errStr;
+  auto invalid = str.find_first_not_of("0123456789-+");
+  if (invalid != std::string_view::npos ||
+      invalid == 0) {
+    errStr = "The option value '";
```

Comment:
Above, we have "using std::ostringstream", and then we don't use ostringstream, which I think would be more idiomatic here.